code/game/objects/items/broom.dm

```diff
@@ -68,10 +68,13 @@
  * * user - The user of the pushbroom
  * * A - The atom which is located at the location to push atoms from
  */
-/obj/item/pushbroom/proc/sweep(mob/user, atom/A)
+/obj/item/pushbroom/proc/sweep(mob/user, atom/atom)
 	SIGNAL_HANDLER
 
-	var/turf/current_item_loc = isturf(A) ? A : A.loc
+	do_sweep(src, user, atom, user.dir)
+
+/proc/do_sweep(broomer, mob/user, atom/atom, dir)
```

Comment:
1. Autodoc
2. Type for `broomer`
3. A more detailed name than `atom/atom` would be nice
4. `dir` could probably be prefixed to be more readable, `sweep_dir`
5. You pass `dir` but `new_item_loc` still gets it from `user.dir`